### DISCUSSION TITLE
PopUntil intop a different navigation stack

### DISCUSSION THREAD

**Discussion by Roboroads** [Milad-Akarie/auto_route_library#1939] [Q&A]
Lets say I have this route stack, where every directory is a nested navigation stack:

- /
- /someform/step1
- /someform/step2
- /someform/step3
- /answer/details

_note here: someform and answer are nested navigation routes_

I want to go back from `/answer/details` all the way to `/someform/step1`. Is there an easy way to router.popUntilNamed to this page? Currently I'm passing the context of /someform to the /answer/details page so I can pop all pages from the answer stack, then popUntilNamed to the step1 page on the someform stack, but this feels hacky.

**Comment by Milad-Akarie** [comment]
@Roboroads have you tried `router.navigate(TargetRoute())`, this will pop all pages until target route if it's already in the stack that is, otherwise it pushes it to the stack.
you can also try setting scoped to false inside `popUntilRouteWithName('target', scoped: false);`

**Reply by Roboroads** [reply]
Thanks! We'll be trying these :)
